### PR TITLE
Fix `x509-cert` crate version in `mc-sgx-dcap-types`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Previously `mc-sgx-dcap-types` specified `x509-cert` at version `0.2.0`.
This version is too low since `mc-sgx-dcap-types` uses the function
[`load_pem_chain`](https://docs.rs/x509-cert/latest/x509_cert/certificate/struct.CertificateInner.html#method.load_pem_chain)
which was added in `0.2.3`.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #346** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/346" target="_blank">Graphite</a> 
    * **PR #349** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/349" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/349?utm_source=stack-comment).

**Review by dolanbernard — APPROVED** [review]

File: dcap/types/Cargo.toml [hunk 1]
```diff
@@ -30,7 +30,7 @@ serde = { version = "1.0.152", default-features = false, features = ["derive"],
 sha2 = { version = "0.10.6", default-features = false }
 static_assertions = "1.1.0"
 subtle = { version = "2.4.1", default-features = false }
-x509-cert = { version = "0.2.0", default-features = false, features = ["pem"], optional = true }
+x509-cert = { version = "0.2.3", default-features = false, features = ["pem"], optional = true }
```
Comment (dolanbernard) [inline dcap/types/Cargo.toml:33]: Any reason this isn't updated in dev-dependencies below as well?

Comment (nick-mobilecoin) [reply dcap/types/Cargo.toml:33]: good catch!, will fix that

Comment (nick-mobilecoin) [reply dcap/types/Cargo.toml:33]: updated the version in the dev-dependencies

**Review by dolanbernard — APPROVED** [review]

**Pull request merged**